actionpack/lib/action_controller/test_case.rb

```diff
@@ -659,7 +659,9 @@ def process(action, *args)
         @request.assign_parameters(@routes, controller_class_name, action.to_s, parameters)
 
         @request.session.update(session) if session
-        @request.flash.update(flash || {})
+
+        is_request_flash_enabled = @request.respond_to?(:flash)
```

Comment:
I moved the `enable_flash` call to an ensure block. Thank you bery much.
But, This `@request.respond_to?(:flash)` change was needed.